Src/Couchbase.Linq/BucketContext.cs

```diff
@@ -32,13 +30,16 @@ public class BucketContext : IBucketContext, IChangeTrackableContext
         /// </summary>
         public bool ChangeTrackingEnabled { get { return _beginChangeTrackingCount > 0; } }
 
+        public IDataStore DataStore { get; private set; }
+
         /// <summary>
         /// Creates a new BucketContext for a given Couchbase bucket.
         /// </summary>
         /// <param name="bucket">Bucket referenced by the new BucketContext.</param>
         public BucketContext(IBucket bucket)
         {
             _bucket = bucket;
+            DataStore = new BucketDataStore(bucket);
```

Comment:
@jeffrymorris 

I'm not sure that I totally agree with the idea of it becoming more complicated.  I think it actually simplifies things by providing a separation of concerns.  Core features are provided in BucketContext, and work by access the IBucket directly.  Advanced direct-access features are in IBucketDataStore, which in turn directly access the IBucket.

I think it also makes testing simpler.  If you're trying to mock IBucketContext, you can pretty easily choose to just not mock the IBucketDataStore if the code you're testing doesn't use the advanced direct-access features like Execute.  But if moved the Execute function into IBucketContext it's then harder to see that separation of concerns and not mock the functions.  

I do agree about the QueryRequest, though.

Brant